### PR TITLE
Update core from Colorbleed repo

### PR DESCRIPTION
**Pull request by BigRoy**
This is a lengthy PR, but contains some good stuff.

**Changes**:
- Fixes: https://github.com/getavalon/core/issues/313
- Adds `switch` to the api. (originally introduced here: https://github.com/getavalon/core/pull/312)
- Adds `switch` support to the the scene inventory (manager)
- Adds InventoryAction (to allow building custom actions for the Scene Inventory for a specific host), see: https://github.com/Colorbleed/core/pull/19
- Allow default family filter state to be set in the studio config, see: https://github.com/Colorbleed/core/pull/21/files
- Improve filtering for scene inventory, see: https://github.com/Colorbleed/core/pull/18
- Also supports custom container metadata in the studio config, see: https://github.com/Colorbleed/core/pull/20 and https://github.com/getavalon/core/pull/321/files#diff-a8cb4fc609f49449f7c3a96ded875830R79 - with this you could for example give custom colors to entries in the Scene Inventory.

### CODE REVIEW

**Comment by mottosso** [comment]
There's too much to review; I say let's merge this as-is, and work out issues that arise from the other end.

**Comment by mottosso** [comment]
If someone could have a look at the conflicts, I think we should merge this and tackle any issues we find afterwards.

**Comment by tokejepsen** [comment]
I may have messed up on the commits but I tried to merge the conflicts [here](https://github.com/Colorbleed/core/pull/22).

File: setup/fusion/scripts/Comp/avalon/publish.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    if settings.ContextLabel == settings.ContextLabelDefault:
+        settings.ContextLabel = "Fusion"
+    if settings.WindowTitle == settings.WindowTitleDefault:
+        settings.WindowTitle = "Pyblish (Fusion)"
+
+
+def _set_current_working_dir():
+    # Set current working directory next to comp
+
+    filename = comp.MapPath(comp.GetAttrs()["COMPS_FileName"])
```
[the remark this answers is not in the document]
Comment (BigRoy) [reply setup/fusion/scripts/Comp/avalon/publish.py]: Unfortunately this is how a Fusion Comp script works, there is a magical `comp` variable declared that you can use in the script. :) So this is intentional.

[the remark this answers is not in the document]
Comment (mottosso) [reply setup/fusion/scripts/Comp/avalon/publish.py]: Is this a global variable provided by Fusion?

If so, we can make it explicit to readers via something like this at the top..

```python
try:
  # Provided by Fusion
  comp
except NameError:
  comp = None
```

[the remark this answers is not in the document]
Comment (mottosso) [reply setup/fusion/scripts/Comp/avalon/publish.py]: I think so, yes. It'd quiet a future developer wondering about it, and the Hound too.

**Comment by BigRoy** [comment]
This should cover it! :)

**Comment by BigRoy** [comment]
Fixed - not sure why the quotations mattered, but now it passes! :)

**Pull request merged**